### DISCUSSION TITLE
How to install a 'subpackage' with Poetry?

### DISCUSSION THREAD

**Discussion by antusystem** [python-poetry/poetry#5124] [Q&A] [answered]
HI, I have a problem installing a 'subpackage' (don't know their real name) but I don't know how. I want to install Pylama because it has other Linting packages. I'm using Poetry for Package Managment, but I don't know how to do this.

Pylama's documents indicate that if you use pip you have to use `pip install pylama[all]`, but I can't find how to do it with Poetry. I have this in my pyproject.toml:

```
[tool.poetry.dependencies]
 python = "^3.8"
 Pillow = "^9.0.0"
 matplotlib = "^3.5.1"

[tool.poetry.dev-dependencies]
 pylama[all] = "^8.3.7"
```
With that I only install the basic from pylama, the same as doing `pip install pylam`a.

Afterwards, I created a virtual environment with virtualenv and use `pip install pylama[all]`, and I it showed this error: `ERROR: pylint 2.12.2 has requirement mccabe<0.7,>=0.6, but you'll have mccabe 0.7.0 which is incompatible`. If I do pip freeze after the installation, I can see pylint==2.12.2. I don't know if it is because of this error that Poetry don't install all, but in that case I guess it should install mypy, eradicate, radon, and vulture, but nope.

I'm running this in Lubuntu, with Python 3.8.10

Hope someone can help me, thanks.

**Comment by clintonroy** [comment, marked as answer]
poetry install pylama[all] does install the extras package named "all" for pylama, leading to a pyproject.toml off:

pylama = {extras = ["all"], version = "^8.3.7"}

Please stick to one package manager per virtual environment, either use pip or poetry.

**Reply by antusystem** [reply]
Ohh, thanks I was writing it with `poetry init`, that way I did knew how to do it. Your explanation worked perfectly. And don't worry, I was trying pip just to compare the packages, although, I am thinking of using conda to control the environment and Poetry to control the packages, maybe using `conda` to install the dev dependencies, but it might be better to let Poetry handle all the dependencies and conda only the python version in the environment. Thanks for your help
